src/service.ts

```diff
@@ -1,116 +1,27 @@
-    }
-}
-
-export const getSnapshotHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateSnapshotsEvent),
-        map(event => event.pathParameters!.term),
-        chain(getSnapshot(es)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
-
-export const getCheckpointBlocksHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateCheckpointBlocksEvent),
-        map(event => event.pathParameters!.term),
-        chain(getCheckpointBlock(es)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
-
-export const getTransactionsHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateTransactionsEvent),
-        map(event => event.pathParameters!.term),
-        chain(getTransaction(es)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
-
-export const getTransactionsBySnapshotHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateTransactionsEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionBySnapshot(es)(term, limit, searchAfter)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
-
-export const getTransactionsByAddressHandler = (event: APIGatewayEvent, es: Client) =>
-    pipe(
-        taskEither.of<ApplicationError, APIGatewayEvent>(event),
-        chain(validateAddressesEvent),
-        map(extractParameters),
-        chain(({term, limit, searchAfter}) => getTransactionByAddress(es)(term, null, limit, searchAfter)),
-        fold(
-            reason => task.of(errorResponse(reason)),
-            value => task.of(successResponse(StatusCodes.OK)(value))
-        )
-    )
-
-export const getTransactionsBySenderHandler = (event: APIGatewayEvent, es: Client) =>
+import { Client } from '@opensearch-project/opensearch'
+import { APIGatewayEvent } from 'aws-lambda'
+import { pipe } from 'fp-ts/lib/pipeable'
+import { task } from "fp-ts/lib/Task"
+import { chain, fold, map, taskEither } from 'fp-ts/lib/TaskEither'
+import { ApplicationError, errorResponse, StatusCodes, successResponse } from './http'
+import { getBalanceByAddress } from './opensearch'
+import { validateAddressesEvent } from './validation'
+
+export const getBalanceByAddressHandler = (event: APIGatewayEvent, es: Client) =>
```

Comment:
you changed `es` to `os` almost everywhere, so please be consistent here as well